code/iaas/logic/src/main/java/io/cattle/platform/process/snapshot/SnapshotCreate.java

```diff
@@ -0,0 +1,90 @@
+package io.cattle.platform.process.snapshot;
+
+import static io.cattle.platform.core.model.tables.SnapshotStoragePoolMapTable.SNAPSHOT_STORAGE_POOL_MAP;
+import io.cattle.platform.core.constants.SnapshotConstants;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.Snapshot;
+import io.cattle.platform.core.model.SnapshotStoragePoolMap;
+import io.cattle.platform.core.model.StoragePool;
+import io.cattle.platform.core.model.Volume;
+import io.cattle.platform.core.model.VolumeStoragePoolMap;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.lock.LockManager;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.process.base.AbstractDefaultProcessHandler;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Named
+public class SnapshotCreate extends AbstractDefaultProcessHandler {
```

Comment:
Extend agentbasedprocesshandler to do the snapshot create at in the orchestration of the snapshot create.